require.js

```diff
@@ -794,6 +822,8 @@
             return require;
         }
 
+        config.requireForId = requireForId = memoize(makeRequire,config.requireById);
```

Comment:
This doesn't make sense, `config.requireById` looks like it's supposed to be a map but here it becomes a function? Why don't we just make `makeRequire` memoized in the first place?

Reply:
config.requireById is the storage, requireForId is the memoized function using it for lookup

Reply:
Ok. And can we make `makeRequire` memoized instead of adding another function?

Reply:
memoize is the generic way to do so for any function, it's used for multiple ones already